sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java

```diff
@@ -876,4 +904,120 @@ private Predicate createNewCaseFilter(CriteriaBuilder cb, From<?, Case> caze, Da
 
 		return newCaseFilter;
 	}
+
+	public Set<Case> getAllRelevantCases(Person person, Disease disease, Date referenceDate) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Case> cq = cb.createQuery(Case.class);
+		Root<Case> caseRoot = cq.from(Case.class);
+
+		Predicate filter =
+				and(cb, createDefaultFilter(cb, caseRoot), buildRelevantCasesFilter(person, disease, referenceDate, cb, caseRoot));
+		cq.where(filter);
+
+		return new HashSet<>(em.createQuery(cq).getResultList());
+	}
+
+	/**
+	 * Returns a filter that can be used to retrieve all cases with the specified
+	 * person and disease whose report date is before the reference date and,
+	 * if available, whose follow-up  until date is after the reference date,
+	 * including an offset to allow some tolerance.
+	 */
+	private Predicate buildRelevantCasesFilter(Person person, Disease disease, Date referenceDate, CriteriaBuilder cb, Root<Case> from) {
+
+		Date referenceDateStart = DateHelper.getStartOfDay(referenceDate);
+		Date referenceDateEnd = DateHelper.getEndOfDay(referenceDate);
+
+		Predicate filter = and(cb, cb.equal(from.get(Case.PERSON), person), cb.equal(from.get(Case.DISEASE), disease));
+
+		filter = and(
+				cb,
+				filter,
+				cb.lessThanOrEqualTo(
+						from.get(Case.REPORT_DATE),
+						DateHelper.addDays(referenceDateEnd, CaseLogic.ALLOWED_CASE_DATE_OFFSET)));
+
+		filter = and(
+				cb,
+				filter,
+				or(
+						cb,
+						// If the case does not have a follow-up until date, use the last
```

Comment:
Onset date should be considered here, as in other case logic

Reply:
OK